articles/search/search-howto-complex-data-types.md

```diff
@@ -244,6 +244,71 @@ To filter on a complex collection field, you can use a **lambda expression** wit
 
 As with top-level simple fields, simple subfields of complex fields can only be included in filters if they have the **filterable** attribute set to `true` in the index definition. For more information, see the [Create Index API reference](/rest/api/searchservice/create-index).
 
+Azure search has the limitation wherein the complex object in all the collections total in a document cannot exceeed 3000
+
+User wil encounter the below error during indexing when there are complex collection exceeding 3000 limit
+
+“A collection in your document exceeds the maximum elements across all complex collections limit. The document with key '1052' has '4303' objects in collections (JSON arrays). At most '3000' objects are allowed to be in collections across the entire document. Remove objects from collections and try indexing the document again."
+
+In some usecases we might need to add more than 3000 items in a collection .In those cases we can pipe and store it as a string .There is no limitation of number of strings stored in an array in azure search
+
+For example if we need to search for all combination of countries and products and categories if we had stored it as complex object as below the combination cannot exceed 3000
```

Comment:
Consider "For example, it wouldn't be possible to use complex types if the `searchScope` array below had more than 3000 elements"

Reply:
Changed it as suggested Consider, For example, it wouldn't be possible to use complex types if the "searchScope" array below had more than 3000 elements